os/arch/arm/src/armv7-r/arm_assert.c

```diff
@@ -972,5 +980,19 @@ void up_assert(const uint8_t *filename, int lineno)
 	(void)boardctl(BOARDIOC_RESET, 0);
 #endif
 
+#ifdef CONFIG_FAULT_MGR
+	fault_queue = mq_open("fault_queue", O_WRONLY | O_CREAT,  0666, 0);
```

Comment:
We need to check the location where the fault occured. We will go to the fault manager only if the fault occured in the app code. In all other cases, we need to restart the system. Until we plan and resolve this, please keep other changes on hold.

Reply:
Now we are first checking the PC value, if the PC is in kernel context, we are rebooting the board from assert handler only.
If the fault in user context then we will be sending a message to fault manager.